src/ProxyExtensions/Microsoft.OData.ProxyExtensions.Lite/RestShallowObjectFetcher.cs

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.OData.Client;
+
+namespace Microsoft.OData.ProxyExtensions.Lite
+{
+    public class RestShallowObjectFetcher : BaseEntityType
+    {
+        private string _path;
+        private bool _isInitialized;
+
+        public new DataServiceContextWrapper Context
+        {
+            get { return (DataServiceContextWrapper) base.Context; }
+            private set { base.Context = value; }
+        }
+
+        public void Initialize(DataServiceContextWrapper context, string path)
+        {
+            Context = context;
+            _path = path;
+            _isInitialized = true;
+        }
+
+        /// <param name="deferSaveChanges">true to delay saving until batch is saved; false to save immediately.</param>
+        protected Task UpdateAsync<T>(T item, bool deferSaveChanges = false)
+        {
+            if (Context == null) throw new InvalidOperationException("Not Initialized");
+            Context.UpdateObject(item as EntityBase);
+            return FetcherSaveChangesAsync(deferSaveChanges);
+        }
+
+        /// <param name="deferSaveChanges">true to delay saving until batch is saved; false to save immediately.</param>
+        protected Task DeleteAsync<T>(T item, bool deferSaveChanges = false)
+        {
+            if (Context == null) throw new InvalidOperationException("Not Initialized");
+            Context.DeleteObject(item);
+            return FetcherSaveChangesAsync(deferSaveChanges);
+        }
+
+        /// <param name="deferSaveChanges">true to delay saving until batch is saved; false to save immediately.</param>
+        /// <param name="saveChangesOption">Save changes option to control how change requests are sent to the service.</param>
+        protected Task FetcherSaveChangesAsync(bool deferSaveChanges = false, SaveChangesOptions saveChangesOption = SaveChangesOptions.None)
```

Comment:
It's not clear this belongs on the fetcher today... It's not clear that calling SaveChangesAsync on a MessageFetcher should impact the overall model being cached by the client... open issue.